### PR TITLE
update plugin for 1.8.x

### PR DESCRIPTION
**Pull request by msmthng**
i have adopted the plugin to get ready for octoprint >= 1.8.0

can you add me to maintainer in your repo?

thx

martin

### CODE REVIEW

**Comment by Mrnt** [comment]
Hi @msmthng, thanks for stepping up! Took a quick look at your GitHub profile - what kind of Python experience do you have?

**Comment by msmthng** [comment]
Hi, iam working on testautmation with robot framework and python python3 
the last 10 years.
The git is only my private one. All the other stuff I did is in the company 
git.

On octoprint iam new, I have installed it last week,because my printer is 
in the in the cellar and I would like to have some network control on my ffcp2.

I have one question related to the idex stuff. I have compared the data 
from Flashprint gcode file and the serial log which was l equal but the 
printer didn't do the job. Do you have an idea why the behavior is 
different even if the command stack is equal.

Regards

Martin

Am 12. Februar 2023 20:54:56 schrieb Mrnt ***@***.***>:

**Comment by Mrnt** [comment]
> printer didn't do the job
Not sure what you mean - the printer didn't do anything at all or didn't do it right? Did you send the code file to to the SD card ("Upload to SD") or print under Octoprint control ("Upload")?

File: octoprint_flashforge/__init__.py [hunk 1]
```diff
@@ -63,6 +65,13 @@ def __init__(self):
 		self._feature_settings = {
 			'autoUppercaseBlacklist': ['M146']		# LED control requires lowercase r,g,b
 		}
+
+		for setting in self._conn_settings.keys():
```
Comment (Mrnt) [inline octoprint_flashforge/__init__.py:69]: I admit I have not looked at the changes in OctoPrint settings - does this replace the need for setting the default_settings? eg:
`default_settings["serial"] = dict_merge(default_settings["serial"], self._conn_settings)`
or is it doing something else?

**Comment by msmthng** [comment]
I did print using octoprint. Sd card upload is not working, should it work 
with ffcp2? The right extruder was moving but both didn't extrude at the 
end I had to cleen both extruder.

Martin

Am 12. Februar 2023 22:00:33 schrieb Mrnt ***@***.***>:

**Comment by msmthng** [comment]
With 1.8.0 the default settings are read only and you have to use the set 
method if you want to change values. Without this modification the 
establishment is not working anymore, because the overwrite of the hello 
code was not used anymore on establishing the serial connection.

Am 12. Februar 2023 22:18:06 schrieb Mrnt ***@***.***>:

**Comment by msmthng** [comment]
i have recognized, that version 1.8.0 is not sending M27, so octoprint is not recognizing a print started on printer from SD (FFCP2)

**Comment by Mrnt** [comment]
@msmthng Thanks! Just tested your changes this with 1.8 and seems to connect OK. 

re the M27 issue, I'm not seeing that when I upload to SD card, but this may be something different with your printer firmware - perhaps the plugin is not getting the expected response and therefore not sending M27.

**Comment by Mrnt** [comment]
@msmthng I did add you as a maintainer. Let me know if you need help debugging the SD issue.

**Pull request merged**